EasyPost/Utilities/RuntimeInfo.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace EasyPost.Utilities
+{
+    internal static class RuntimeInfo
+    {
+        internal struct ApplicationInfo
+        {
+            /// <summary>
+            ///     Get the version of the application as a string.
+            /// </summary>
+            /// <returns>The version of the application as a string.</returns>
+            internal static string ApplicationVersion
+            {
+                get
+                {
+                    try
+                    {
+                        Assembly assembly = typeof(ApplicationInfo).Assembly;
+                        FileVersionInfo info = FileVersionInfo.GetVersionInfo(assembly.Location);
+                        return info.FileVersion ?? "Unknown";
+                    }
+                    catch (Exception)
+                    {
+                        return "Unknown";
+                    }
+                }
+            }
+
+            /// <summary>
+            ///     Get the .NET framework version as a string.
+            /// </summary>
+            /// <returns>The .NET framework version as a string.</returns>
+            internal static string DotNetVersion
+            {
+                get { return Environment.Version.ToString(); }
+            }
+        }
+
+        internal struct OperationSystemInfo
+        {
+            /// <summary>
+            ///     Get details about the operating system.
+            /// </summary>
+            /// <returns>Details about the operating system.</returns>
+            private static OperatingSystem OperatingSystem
+            {
+                get { return Environment.OSVersion; }
+            }
+
+            /// <summary>
+            ///     Get user-agent pre-formatted operating system details.
+            /// </summary>
+            /// <returns>Operating system details pre-formatted for the User-Agent header.</returns>
+            internal static string UserAgentOsDetails
+            {
+                get { return $"OS/{Name} OSVersion/{Version} OSArch/{Architecture}"; }
```

Comment:
suggestion: I would rather build these directly on the string where the others are getting added instead of abstract these into a function. It makes it more difficult to determine what the final user-agent header will be when reading the code.

Reply:
Resolved